### PR TITLE
Add request-promise@4.1.1 w/ npm auto-update

### PR DESCRIPTION
**Pull request by extend1994**
Pull request for issue: #10517

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/request/request-promise
 * NPM package url (optional): https://www.npmjs.com/package/request-promise
 * License and its reference: [ISC](https://github.com/request/request-promise/blob/master/LICENSE)
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 44
   - Count of stars: 1,433
   - Count of forks: 105
 * NPM download stats (optional):
   - Downloads in the last day: 63,927
   - Downloads in the last week: 326,480
   - Downloads in the last month: 1,268,622

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by extend1994** [comment]
@cdnjs/intern3 Please help me review this pull request, thank you.

**Comment by extend1994** [comment]
@analog-nico Thanks!

**Comment by extend1994** [comment]
ping @clairetsai818 @sufuf3

File: ajax/libs/request-promise/package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "url": "git+https://github.com/request/request-promise.git"
+  },
+  "author": "Nicolai Kamenzky (https://github.com/analog-nico)",
+  "license": "ISC",
+  "npmName": "request-promise",
+  "npmFileMap": [
+    {
+      "basePath": "lib",
+      "files": [
+        "rp.+(js|css|map)"
```
Comment (sufuf3) [inline ajax/libs/request-promise/package.json]: Use glob.sync('rp.+(js|css|map)') just got `rp.js`.

Comment (sufuf3) [inline ajax/libs/request-promise/package.json]: Use glob.sync('rp.+(js|css|map)') just got `rp.js`.

**Comment by extend1994** [comment]
@sufuf3 I modified the rule to make it more flexible. But the old one `rp.+(js|css|map)` works for current version cause there is only one file `rp.js` in the npm. Please review it again, thanks.

**Comment by PeterDaveHello** [comment]
Any updates here? Thanks!

**Comment by extend1994** [comment]
@PeterDaveHello Sorry about missing it!
Here is the author's reply: request/request-promise#180.
@pvnr0082t Could you please help me check the reply. I am not sure that if this library is compatible with browser after receiving the reply? Thanks!

**Comment by PeterDaveHello** [comment]
I replied.

**Comment by extend1994** [comment]
Thanks! :+1:

**Comment by sufuf3** [comment]
@PeterDaveHello @extend1994 as the author said he need for request/request-promise#93  to get done, then will release browserified file.
But the process is slow.

Should we still open this PR until the upstream provide the file?
Or close this PR and mention in #10517?

Thank you.

**Comment by PeterDaveHello** [comment]
I think we can reopen this once it's updated from upstream, but just temporarily close this first.

**Comment by PeterDaveHello** [comment]
@extend1994 you should subscribe to https://github.com/request/request-promise/issues/93

**Pull request closed without merging**